lib/fli/FliObjHdl.cpp

```diff
@@ -0,0 +1,530 @@
+    return -1;
+}
+
+int FliValueObjHdl::set_signal_value(const long value)
+{
+    LOG_ERROR("Setting signal/variable value via long not supported for %s of type %d", m_fullname.c_str(), m_type);
+    return -1;
+}
+
+int FliValueObjHdl::set_signal_value(std::string &value)
+{
+    LOG_ERROR("Setting signal/variable value via string not supported for %s of type %d", m_fullname.c_str(), m_type);
+    return -1;
+}
+
+int FliValueObjHdl::set_signal_value(const double value)
+{
+    LOG_ERROR("Setting signal/variable value via double not supported for %s of type %d", m_fullname.c_str(), m_type);
+    return -1;
+}
+
+int FliEnumObjHdl::initialise(std::string &name, std::string &fq_name)
+{
+    m_num_elems   = 1;
+    m_value_enum  = mti_GetEnumValues(m_val_type);
+    m_num_enum    = mti_TickLength(m_val_type);
+
+    return FliValueObjHdl::initialise(name, fq_name);
+}
+
+const char* FliEnumObjHdl::get_signal_value_str(void)
+{
+    if (m_is_var) {
+        return m_value_enum[mti_GetVarValue(get_handle<mtiVariableIdT>())];
+    } else {
+        return m_value_enum[mti_GetSignalValue(get_handle<mtiSignalIdT>())];
+    }
+}
+
+long FliEnumObjHdl::get_signal_value_long(void)
+{
+    if (m_is_var) {
+        return (long)mti_GetVarValue(get_handle<mtiVariableIdT>());
+    } else {
+        return (long)mti_GetSignalValue(get_handle<mtiSignalIdT>());
+    }
+}
+
+int FliEnumObjHdl::set_signal_value(const long value)
+{
+    if (m_const) {
+        LOG_ERROR("Attempted to set a constant signal/variable!\n");
+        return -1;
+    }
+
+    if (value > m_num_enum || value < 0) {
+        LOG_ERROR("Attempted to set a enum with range [0,%d] with invalid value %d!\n", m_num_enum, value);
+        return -1;
+    }
+
+    if (m_is_var) {
+        mti_SetVarValue(get_handle<mtiVariableIdT>(), value);
+    } else {
+        mti_SetSignalValue(get_handle<mtiSignalIdT>(), value);
+    }
+
+    return 0;
+}
+
+int FliLogicObjHdl::initialise(std::string &name, std::string &fq_name)
+{
+    switch (m_fli_type) {
+        case MTI_TYPE_ENUM:
+            m_num_elems   = 1;
+            m_value_enum  = mti_GetEnumValues(m_val_type);
+            m_num_enum    = mti_TickLength(m_val_type);
+            break;
+        case MTI_TYPE_ARRAY: {
+                mtiTypeIdT elemType = mti_GetArrayElementType(m_val_type);
+                int buff_len = 0;
+
+                m_ascending   = (mti_TickDir(m_val_type) == 1);
+                m_value_enum  = mti_GetEnumValues(elemType);
+                m_num_enum    = mti_TickLength(elemType);
+                m_num_elems   = mti_TickLength(m_val_type);
+                buff_len      = (m_num_elems + (sizeof(*m_mti_buff) - 1)) / sizeof(*m_mti_buff);
+
+                m_mti_buff    = (mtiInt32T*)malloc(sizeof(*m_mti_buff) * buff_len);
+                if (!m_mti_buff) {
+                    LOG_CRITICAL("Unable to alloc mem for value object mti read buffer: ABORTING");
+                    return -1;
```

Comment:
@stuarthodgson This is just a style I tend to follow, so when looking at the code, even if you don't know what LOG_CRITICAL will do you can tell it will exit at that point.

I can remove it if you want, just let me know.